### PR TITLE
Add ability to selectively disable SQS events

### LINKED ISSUE #43: Stop forwarding "verified" events to /fxa-register/ on basket
As discussed in https://github.com/mozmeao/basket/pull/35/ basket will soon start receiving and processing these events itself, so there's no longer any need for us to translate them into calls to /fxa-register/.

This should be a straightforward little patch so I'm putting my face on it for tomorrow and hoping to make trian-96.

### PR DESCRIPTION
**Pull request by rfk**
Fixes #43.  This turned into a bit of a rabbit-hole, but I think it came up OK.  I recommend reviewing the two commits independently.

In the spirit of converting code as we go, the first commit ES6-ifies ./lib/events/index.js and its corresponding tests.  This was complicated because coverage and linting were using older tools that didn't seem to support ES6 constructs.  I've migrated this repo to use a similar setup to fxa-oauth-server and it seems to work OK.

The second commit is the actual change, allowing us to pref off processing of individual SQS event types while still keeping the code around in case we need to rollback.  The tests were a little hairy because the pref modified global module state, but I didn't want to ugly it up too much for the sake of the tests.

@philbooth do you have bandwidth for an r? here for trian-96?

### CODE REVIEW

**Comment by rfk** [comment]
> This turned into a bit of a rabbit-hole, but I think it came up OK.

And of course, it immediately fails in CI...

File: lib/events/index.js [hunk 1]
```diff
@@ -2,30 +2,37 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-var url = require('url');
-var Promise = require('bluebird');
-var basket = require('../basket');
-var logger = require('../logging')('events');
+const url = require('url');
```
Comment (philbooth) [inline lib/events/index.js]: Also add a `'use strict'` if we're updating all this other stuff?

File: lib/events/index.js [hunk 2]
```diff
@@ -2,30 +2,37 @@
...
-var messageHandlers = {
+const messageHandlers = module.exports._messageHandlers = {
   verified: onVerified,
   login: onLogin
 };
 
+const DISABLED_EVENT_TYPES = config.get('basket.sqs.disabled_event_types');
+if (DISABLED_EVENT_TYPES) {
+  DISABLED_EVENT_TYPES.forEach((typ) => {
+    delete messageHandlers[typ];
```
Comment (philbooth) [inline lib/events/index.js]: I'll withhold my typical objection about  `delete` here because it's just initialisation code.

Comment (rfk) [reply lib/events/index.js]: I'm open to doing this a different way, e.g. we could have one that's the set of all handlers, but copy the non-disabled ones into a separate object for use at runtime.

Comment (philbooth) [reply lib/events/index.js]: Or even just setting to `undefined`, I prefer to `delete` generally. But it's initialisation code, it only gets executed once, so whatever. :smile:

**Comment by rfk** [comment]
After a protracted battle with circle, I emerged the victor; merging...

**Pull request merged**